core/src/View/MDSpan/Kokkos_MDSpan_Layout.hpp

```diff
@@ -146,15 +147,34 @@ KOKKOS_INLINE_FUNCTION auto mapping_from_array_layout_impl(
           extents_type{dextents<index_type, MappingType::extents_type::rank()>{
               layout.dimension[Idx]...}}};
     } else {
-      // This doesn't work because LayoutRight is not the same as
-      // std::layout_right_padded
-      static_assert(!(std::is_same_v<ArrayLayout, LayoutRight> &&
-                      extents_type::rank() > 2));
+// Handle DEFAULT_ARG, should be layout_dimension 0 or n -1
+// assert that this is not default_arg, as a tool for people to
+// transition their code and avoid breaking changes
+#ifdef KOKKOS_ENABLE_IMPL_CHECK_POSSIBLY_BREAKING_LAYOUTS
+      if constexpr (std::is_same_v<ArrayLayout, LayoutRight> &&
+                    extents_type::rank() > 2) {
+        if (layout.stride != KOKKOS_IMPL_CTOR_DEFAULT_ARG) {
+          std::cerr << "The layout of values in this Kokkos View may be "
```

Comment:
Seeing that this is a device callable function, I imagine that we don't want a `std::cerr` in here?

Reply:
oh yikes good catch. Hmm do we have a good way of issuing warnings from device? I suppose our printf but that seems bad.

Reply:
I updated this to be wrapped by `KOKKOS_IF_ON_HOST`